cmd/retrieve.go

```diff
@@ -96,6 +113,81 @@ func Retrieve(cmd *cobra.Command, _ []string) (err error) {
 		filter.PageNames = pageNames
 	}
 
+	var sinceStr string
+	since := time.Now()
+	hasSince := false
+	if sinceStr, err = cmd.Flags().GetString(flags.Since.Name); err != nil {
+		return err
+	} else if len(sinceStr) > 0 {
+		// First try to parse something like "01/02 03:04:05PM '06 -0700"
+		if parseTry, err := time.Parse(time.Layout, sinceStr); err == nil {
+			hasSince = true
+			since = parseTry
+		}
+		// Next try to parse with the other layout constants in time
+		if !hasSince {
```

Comment:
![image](https://user-images.githubusercontent.com/30167939/231274270-6b26fcdd-c85f-4896-8df6-c05e9c33936f.png)
After this condition, the timezone is lost

Reply:
I'm getting the same error I reported initially because since.UTC() is not working because the reference of the timezone is lost on that condition

Reply:
What value are you providing to `--since`? I can't see in this screenshot, but I suspect you're providing a value for `since` that does not have a timezone specified, and `time.Parse()` is assuming in that case that the value is *already* in UTC.

It would probably be more the intended behavior if, before `time.Parse()`, we assign the "local" time zone of the computer. But only if the time zone is not specified.

Reply:
go run main.go deploy --host='since6.pliny.webserver:3000'  -d /Users/gzapatas/Downloads/since6 -u='gzapatas' -p='12345678' --since="2023-04-11 14:49:00" -v

I'm running that without the timezone

Reply:
@acofer Pliny and warden are working ok for this ticket, skuid-cli works with since, app, and both.

I just have a few comments before approving it:

- If I set a future date, the retrieve gets the entire site (Are you going to fix the errors on this ticket or should we open a new one? It is supposed that everyone set an old date)
- There are some failing formats (Are you going to fix the errors on this ticket or should we open a new one? Because those formats are not really common)

Layout      = "04/14 02:04:05PM '23 -0500" -> Ok Since 2023-04-14T19:04:05Z
ANSIC       = "Fri Apr 14 14:04:05 2023"  -> Ok Since 2023-04-14T19:04:05Z
**UnixDate    = "Fri Apr 14 14:04:05 MST 2023" -> Error Since 1996-05-26T02:34:27Z**
RubyDate    = "Fri Apr 14 14:04:05 -0500 2023" -> Ok Since 2023-04-14T19:04:05Z
**RFC822      = "14 Apr 23 14:04 MST" -> Error Since 2023-04-14T14:04:00Z (Should be 14 Apr 23 21:04 UTC?)**
RFC822Z     = "14 Apr 23 14:04 -0500" -> Ok Since 2023-04-14T19:04:00Z
**RFC850      = "Friday, 14-Apr-23 14:04:05 MST" -> Error Since 2023-04-14T14:04:05Z (Should be 14 Apr 23 21:04 UTC?)
RFC1123     = "Fri, 14 Apr 2023 14:04:05 MST" -> Error Since 2023-04-14T14:04:05Z (Should be 14 Apr 23 21:04 UTC?)**
RFC1123Z    = "Fri, 14 Apr 2023 14:04:05 -0500" -> Ok Since 2023-04-14T19:04:05Z
**RFC3339     = "2023-04-14T14:04:05Z05:00" -> Error "A --since option was provided but was not parsable as a time or duration"
RFC3339Nano = "2023-04-14T14:04:05.999999999Z05:00" -> Error "A --since option was provided but was not parsable as a time or duration"**
Kitchen     = "2:04PM" -> Ok Since 2023-04-14T19:04:05Z
Stamp      = "Apr 14 14:04:05" -> Error Since 2023-07-27T19:04:05Z
**StampMilli = "Apr 14 14:04:05.000" -> Error Since 2023-07-27T19:04:05Z
StampMicro = "Apr 14 14:04:05.000000" -> Error Since 2023-07-27T19:04:05Z
StampNano  = "Apr 14 14:04:05.000000000" -> Error Since 2023-07-27T19:04:05Z**
DateTime   = "2023-04-14 14:04:05" -> Ok Since 2023-04-14T19:04:05Z
**DateOnly   = "2023-04-14" -> Ok Since 2023-04-14T05:00:00Z (Comment Are you setting 05:00:00?)**
TimeOnly   = "14:04:05" -> Ok Since 2023-04-14T19:04:05Z
1s, 1m, 1h, 1d, 1M, 1y -> Ok all of the work as expected

Reply:
* For the "future" thing, I'm not sure why you get the entire site rather than nothing at all, but better UX would be to print a warning to the console and exit - users should not put in a future date on purpose. I'll fix it.
* I'll check on the date parsing cases you listed. It's odd that RFC3339 did not work; that was the format I preferred and used to test with. Note that these formats seem to be very strict as far as the regular expression matching. (But "fuzzy" date parsing is a big lift if there isn't already open-source out there we can use - it think it would take a long time to write something that is useable.)

As for the time zone offsets - **sigh**. I thought Go's `time.ParseInLocation()` would read "MST" and actually, you know, parse that time as MST time zone. The `time` library seems to be a lot dumber than I assumed it was.

Reply:
Sooo... The only time zone abbreviation in, like, RFC822 that I can get to reliably parse is "EST"/"EDT". The rest of them (so far have tried "MST"/"MDT" and "CST"/"CDT") get parsed as though they are already in UTC (offset +00:00). 🤦 I'm trying to figure out what's wrong.

Reply:
Looks like a Go bug... https://stackoverflow.com/questions/39539210/why-go-is-parsing-my-timestamp-with-local-location-instead-of-utc 

The only locations it seems to understand are your current system local location and UTC, even if you `import _ time/tzdata`. I'm going to have to think about a way around this, or just remove the "MST"-style formats from possible parse layouts to avoid confusion. Man, this is terrible. Thanks for helping me find the issues and work through them.